scripts/create-clusters.sh

```diff
@@ -113,10 +113,16 @@ function create-clusters() {
     unset KUBECONFIG
   fi
 
-  # TODO(font): Configure insecure registry on kind host cluster. Remove once
-  # https://github.com/kubernetes-sigs/kind/issues/110 is resolved.
-  echo "Configuring insecure container registry on kind host cluster"
-  configure-insecure-registry-on-cluster 1
+  # Avoid configuring the insecure registry unless explicitly
+  # configured to do so. If the docker daemon is running on another
+  # host (e.g. docker running in a VM on MacOS) the attempt to
+  # configure the registry on the local host may fail.
+  if [[ "${CONFIGURE_INSECURE_REGISTRY}" ]]; then
```

Comment:
This flag was originally intended to configure the insecure registry on the host, which is different from configuring it in the cluster and should be mutually exclusive. That is, you may already have your host configured for an insecure registry already, and do not want the script touching your config, but the clusters' docker config still needs to be set up with the insecure registry option.

If we really need to make it an option to configure an insecure registry in the cluster, I suggest we use a new option e.g. `CONFIGURE_INSECURE_REGISTRY_CLUSTER` and possibly rename this option to `CONFIGURE_INSECURE_REGISTRY_HOST`.

Also, can we turn this on by default? I think if we're doing dev testing, it doesn't hurt to always set up the clusters with an insecure registry config regardless if you end up using it. However, I think we're more than likely to want to use one, even if we don't want the script to automatically configure the host for an insecure container registry (by modifying your config file) either because you don't trust the script or you've already previously set it up. In any case, I'm biased because I always use this. :smile: I'm not sure about others.

Reply:
This step was failing because kind 0.3.0 switched to containerd. I've added #964 to ensure we update at some point.  For now I've added `CONFIGURE_INSECURE_REGISTRY_HOST` as per your suggestion and defaulted it to true.

My testing workflow involves deploying once and then running in-memory, so the local registry doesn't buy me much and after the ci configuration problem I stopped using it entirely.

Reply:
Ah, ok. Yes we should switch to 0.3.0 at some point in the near future which may make these comments moot.